distributed/utils_test.py

```diff
@@ -2100,8 +2099,10 @@ def raises_with_cause(
     match: str | None,
     expected_cause: type[BaseException] | tuple[type[BaseException], ...],
     match_cause: str | None,
-) -> Generator[None, None, None]:
-    """Contextmanager to assert that a certain exception with cause was raised
+    *more_causes: type[BaseException] | tuple[type[BaseException], ...] | str | None,
+) -> Iterator[None]:
```

Comment:
type should be Generator as contextmanager calls .throw

It's a bug that contextmanager accepts Iterator functions

Reply:
Today I learned somthing new :)
- https://github.com/python/typeshed/issues/2772